com.avaloq.tools.ddk.checkcfg.ui/src/com/avaloq/tools/ddk/checkcfg/ui/templates/CheckCfgTemplateProposalProvider.java

```diff
@@ -216,31 +218,29 @@ public String apply(final ConfiguredCheck from) {
       }
     }), Predicates.notNull());
     final CheckCatalog catalog = configuredCatalog.getCatalog();
+    final JvmType stringType = typeRefBuilderFactory.create(context.getResource().getResourceSet()).typeRef(String.class).getType();
     for (final Check check : catalog.getAllChecks()) {
       // create a template on the fly
       final String checkName = check.getName();
       if (!Iterables.contains(alreadyConfiguredCheckNames, checkName)) {
 
         // check if referenced check has configurable parameters
-        String paramString = ""; //$NON-NLS-1$
-        if (!check.getFormalParameters().isEmpty()) {
-          StringBuilder params = new StringBuilder("("); //$NON-NLS-1$
-          for (final FormalParameter p : check.getFormalParameters()) {
-            final String paramName = p.getName();
-            final String defaultValue = String.valueOf(interpreter.evaluate(p.getRight()).getResult());
-            params.append(paramName).append(" = ").append("${").append(defaultValue).append('}'); //$NON-NLS-1$ //$NON-NLS-2$
-            params.append(", "); //$NON-NLS-1$
-          }
-          if (params.length() > 2) {
-            paramString = params.substring(0, params.length() - 2) + ')';
-          }
+        final StringJoiner paramsJoiner = new StringJoiner(", ", " ( ", ")"); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+        paramsJoiner.setEmptyValue(""); //$NON-NLS-1$
+        for (final FormalParameter param : check.getFormalParameters()) {
+          final JvmType paramType = param.getType().getType();
+          final String paramName = param.getName();
+          final String defaultValue = String.valueOf(interpreter.evaluate(param.getRight()).getResult());
+
+          // Use SimpleEnumTemplateVariableResolver with strings because otherwise TemplateTranslator can't cope with whitespace
+          paramsJoiner.add(paramName + " = " + (Objects.equals(stringType, paramType) ? "\"${defaultValue:SimpleEnum('" + defaultValue + "')}\"" //$NON-NLS-1$//$NON-NLS-2$ //$NON-NLS-3$
+              : "${" + defaultValue + '}')); //$NON-NLS-1$

```

Comment:
Whatever `SimpeEnum` does here, this whole part looks awfully weird. Can we get rid of it? Already, you're explicitly adding the quotes, what does `SimpleEnum` even do, and why is it bad to do it for the non-string case?